File: solutions/beecrowd/1581/1581.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    std::int16_t n, k;
    std::string language, previous_language;
    bool all_languages_equal;

    std::cin >> n;

    while (n--) {
        std::cin >> k;
        std::cin >> previous_language;

        all_languages_equal = true;

        k--;
        while (k--) {
            std::cin >> language;

            if (previous_language != language) {
                all_languages_equal = false;
            }
        }

        std::cout << (all_languages_equal ? language : "ingles") << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1581/1581.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

n = int(input())

for _ in range(n):
    # minus 1, because we read one langage before for loop
    k = int(input()) - 1
    language = input().strip()
    previous_language = language
    all_languages_equal = True

    for _ in range(k):
        language = input().strip()

        if previous_language != language:
            all_languages_equal = False

    output(f'{language if all_languages_equal else "ingles"}\n')
